cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -169,6 +182,93 @@ protected void setMetricsEnabled(boolean isMetricsEnabled) {
         this.isMetricsEnabled = isMetricsEnabled;
     }
 
+    protected String getLogConfig() {
+        return logConfig;
+    }
+
+    private Map getDefaultLogConfig() {
+        Map<String, String> defaultSettings = new LinkedHashMap<>();
+        // default values
+        defaultSettings.put("log4j.appender.CONSOLE", "org.apache.log4j.ConsoleAppender");
+        defaultSettings.put("log4j.appender.CONSOLE.layout", "org.apache.log4j.PatternLayout");
+        defaultSettings.put("log4j.appender.CONSOLE.layout.ConversionPattern", "%d{ISO8601} %p %m (%c) [%t]%n");
+        if (name.contains("zookeeper")) {
+            defaultSettings.put("zookeeper.root.logger", "INFO");
+            defaultSettings.put("log4j.rootLogger", "${zookeeper.root.logger}, CONSOLE");
+        } else if (name.contains("kafka")) {
+            defaultSettings.put("kafka.root.logger.level", "INFO");
+            defaultSettings.put("log4j.rootLogger", "${kafka.root.logger.level}, CONSOLE");
+            defaultSettings.put("log4j.logger.org.I0Itec.zkclient.ZkClient", "INFO");
+            defaultSettings.put("log4j.logger.org.apache.zookeeper", "INFO");
+            defaultSettings.put("log4j.logger.kafka", "INFO");
+            defaultSettings.put("log4j.logger.org.apache.kafka", "INFO");
+            defaultSettings.put("log4j.logger.kafka.request.logger", "WARN, CONSOLE");
+            defaultSettings.put("log4j.logger.kafka.network.Processor", "OFF");
+            defaultSettings.put("log4j.logger.kafka.server.KafkaApis", "OFF");
+            defaultSettings.put("log4j.logger.kafka.network.RequestChannel$", "WARN");
+            defaultSettings.put("log4j.logger.kafka.controller", "TRACE");
+            defaultSettings.put("log4j.logger.kafka.log.LogCleaner", "INFO");
+            defaultSettings.put("log4j.logger.state.change.logger", "TRACE");
+            defaultSettings.put("log4j.logger.kafka.authorizer.logger", "WARN");
+        }
+        return defaultSettings;
+    }
+
+    private String createLogConfigMap(Map<String, Object> newSettings) {
+        StringBuilder sb = new StringBuilder();
```

Comment:
Again, one of the interesting questions is why not use `Properties` class here? I probably already forgot how much I hated it when I used it for the configuration fields :-D.